dashboard/test/ui/features/step_definitions/blockly.rb

```diff
@@ -246,3 +246,25 @@ def current_block_xml
     return __TestInterface.getBlockXML();
   JS
 end
+
+def clear_main_block_space
+  # Do our async wait on the JavaScript side instead of polling over the wire.
+  # See https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/JavascriptExecutor.html
+  # and https://www.rubydoc.info/gems/selenium-webdriver/3.8.0/Selenium/WebDriver/Driver#execute_async_script-instance_method
+  result = @browser.execute_async_script <<-JS
```

Comment:
We haven't used `execute_async_script` before.  It passes a callback as the last argument to your JavaScript, and blocks until the callback is called (or the selenium script timeout is reached).

The idea here is that we don't need our test runner to poll SauceLabs waiting for something to be available - we can do that in injected JavaScript on the client and call back when ready.   More generally, I wish we used fewer over-the-wire commands per Cucumber step, and I think this is a step in the right direction.